Objects/abstract.c

```diff
@@ -733,6 +734,41 @@ int PyObject_CopyData(PyObject *dest, PyObject *src)
     return 0;
 }
 
+int
+PyObject_CopyToObject(PyObject *obj, void *buf, Py_ssize_t len,
+                      char Py_UNUSED(fort))
+{
+    Py_buffer view_obj;
+
+    if (!PyObject_CheckBuffer(obj)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "destination must be bytes-like objects");
+        return -1;
+    }
+
+    if (!buf) {
+        PyErr_SetString(PyExc_MemoryError, "buf pointer is null");
```

Comment:
I don't think  this should set this exception. `buf` should not be NULL at all, so it should probably raise some fatal error. In addition this check could be performed before the check buffer since it's cheaper.

Reply:
I think that's fine.

Reply:
Please tell me why it's fine? it is important to raise exceptions of suitable types. Here, we don't have a memory error. The issue is the invalid argument being passed. I would either use an assertion (and you should also document that the source buffer must not be NULL) or use `PyErr_BadArgument()`.

Reply:
Maybe I'm thinking too much. When the user really accidentally passes a `NULL`, a segmentation fault will occur when using `memcpy`.
But it's okay, I've made the change.

Reply:
If you don't want the check, I recommend adding an assertion then. Because this is a public function and we really don't want users to misuse it and have a random segfault. Segfault are harder to locate than assertions (and you would also save work).

Ideally, the assertion can be put at the very top of the function.

Reply:
I thought it would be a good idea to add an assertion, so I decided to put it on the first line.

Reply:
I have modified